Automattic/pocket-casts-ios
# Ratings: Pulls ratings in from the API and displays them if they're available

## Description
This integrates with the ratings API to display ratings for podcasts if they're available. 

I also added a view helper items:
1. A `ThemedHostingController` that lets you wrap Swift UI views with the theme environment setup
2. A `ModifedHostingController` which lets you modify the content of a SwiftUI view and pass it easily to a hosting controller. This lets you set custom environments and do other modifications
3. A View extension that maps a SwiftUI view to a UIView
4. A generic JSONDecodableURLTask that lets you convert basic JSON responses without needing to implement all the network code again

## Screenshots
| Example | Example | Example |
|:---:|:---:|:---:|
|<img src="https://user-images.githubusercontent.com/793774/226089501-b371496d-1e57-49c9-ad13-9ed2585a2529.PNG" width="320" />|<img src="https://user-images.githubusercontent.com/793774/226089504-f15737f5-81c2-4652-a3bd-1cf26f7d81a2.PNG" width="320" />|<img src="https://user-images.githubusercontent.com/793774/226089499-4d76c0ae-425a-4070-a9d6-72876c5b59a2.PNG" width="320" />|

## To test

1. Launch the app
5. Navigate to a podcast
6. ✅ Verify nothing displays and the app doesn't crash
7. Enable the showRatings feature flag
8. Navigate to a podcast again, ✅ Verify nothing displays and the app doesn't crash
9. Since the API isn't available yet you can do the following to test:
10. Open RetrievePodcastRatingTask
11. Replace the `endpoint` contents with `"https://podcast.emily.tools/ratings/test.php?total=1234&average=4.32"`
12. This will generate the same response but without loading from the db
13. You can change the total and average numbers to get different values
14. ✅ Verify the star rating appears

## Checklist
